core/shared/src/main/scala/coursier/util/Tree.scala

```diff
@@ -1,57 +1,74 @@
 package coursier.util
 
-import scala.collection.mutable
-import scala.collection.mutable.ArrayBuffer
+import scala.annotation.tailrec
 
-object Tree {
+private[util] object Tree {
+
+  def apply[T](roots: Seq[T])(children: T => Seq[T],
+                              show: T => String): String = {
```

Comment:
I believe you're right. Feel free to open a minimised reproduction issue against MiMa.